### ISSUE TITLE
incorrect molecule when given vague species name

### ISSUE THREAD

**Issue by mjohnson541** [ReactionMechanismGenerator/RMG-website#144]
When I searched for "ketohydroperoxide" it returned the molecule with the smiles string OOOOO.  Now I've seen ketohydroperoxide refer to both O=CCCOO and O=COCOO, but the smiles it returns clearly does not have a ketone on it so it can't be a ketohydroperoxide.  OOOOO also doesn't appear to have any names associated with it.

**Comment by mliu49** [comment]
We use the [NCI resolver](https://cactus.nci.nih.gov/chemical/structure) for processing chemical names. In general, searching by chemical name is not very reliable. You could consider sending them a bug report, but I don't know how responsive they are.

**Comment by mjohnson541** [comment]
Couldn't we just error if the species identifier searched for isn't the smiles string or in the list of names returned by the resolver for that identifier?

**Comment by rwest** [comment]
1. if you visit https://cactus.nci.nih.gov/chemical/structure/ketohydroperoxide/smiles you get the SMILES string, which is what our website does behind the scenes. If you append `/xml` to the URL and visit https://cactus.nci.nih.gov/chemical/structure/ketohydroperoxide/smiles/xml then you get additional information, revealing how it was parsed. Sometimes (eg. https://cactus.nci.nih.gov/chemical/structure/methyl%20formate/smiles/xml ) there will be results from two or more parsers available this way.
2. In this case it interpreted the name not via a lookup (ketohydroperoxide is not in their database of 68million names?), but via "OPSIN" (`resolver="name_by_opsin"`). 
3. OPSIN is the IUPAC name parser from the chemistry department in Cambridge: http://opsin.ch.cam.ac.uk/
4. My guess is it's interpreting it using a Radicofunctional naming convention. But the [IUPAC blue book](http://www.acdlabs.com/iupac/nomenclature/79/r79_786.htm) reveals that the name has not followed the rules, and you should put hydroperoxyl first and end with ketone.  
5. If you try "keto", OPSIN guesses an O atom with two radicals. "keto alcohol" it sticks alcohol (OH) groups on each, getting HOOOH. "keto hydroperoxide" it sticks OOH groups on each, getting HOOOOOH. Which is what you see. It seems "keto" is the same as "oxo". 
6. Try this sequence and it suddenly makes sense:
  - 2 keto propyl hydroperoxide
  - keto propyl hydroperoxide
  - keto ethyl hydroperoxide
  - keto methyl hydroperoxide
  - keto hydroperoxide
7. You could also try the name hydroperoxy methylformate.

Basically, ketohydroperoxide is the wrong name :-D

What do do when someone enters it, is tricky. At least we draw the picture and look up names so they know what they're getting and can detect their mistake.
